core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -218,17 +218,35 @@ oppia.factory('TopicUpdateService', [
           throw Error('A deleted subtopic cannot be restored');
         });
         if (newlyCreated) {
```

Comment:
Does this need to happen before _applyChange? (Is _applyChange asynchronous?)

Reply:
Ah, hm. Could you explain why when you write this before _applyChange, it fails? I think we should understand that.

Reply:
That's because the delete subtopic command is added to the changelist (_applyChange would be below the check), while it's corresponding add subtopic is not there.
Actually, if I just return in that case without calling _applyChange at all, that could also work.